client/src/components/DataFiles/DataFilesToolbar/DataFilesToolbar.test.jsx

```diff
@@ -233,4 +236,105 @@ describe('DataFilesToolbar', () => {
     );
     expect(getByText(/Empty/).closest('button')).toBeDisabled();
   });
+
+  // Test to prevent large file download through TAPIS
+  it('prevents downloads of large files through TAPIS directly', () => {
+    // Mock the alert function
+    global.alert = vi.fn();
+    // Create a test file whose size is greater than 2 GB
+    const testFile = {
+      name: 'test.txt',
+      type: 'file',
+      length: 3000000000,
+      path: '/test.txt',
+    };
+    // Create the store
+    const { getByText } = renderComponent(
+      <DataFilesToolbar scheme="private" api="tapis" />,
+      mockStore({
+        workbench: {
+          config: {
+            extract: '',
+            compress: '',
+            trashPath: '.Trash',
+          },
+        },
+        files: {
+          params: {
+            FilesListing: {
+              system: 'frontera.home.username',
+              path: 'home/username',
+              scheme: 'private',
+            },
+          },
+          listing: { FilesListing: [testFile] },
+          selected: { FilesListing: [0] },
+          operationStatus: { trash: false },
+        },
+        systems: systemsFixture,
+        projects: { metadata: [] },
+        authenticatedUser: { user: { username: 'testuser' } },
+      })
+    );
+    // Click on the download button to try and download the file
+    fireEvent.click(getByText('Download'));
+    // Test for the alert message
+    expect(global.alert).toHaveBeenCalledWith(
+      'The data set that you are attempting to download is too large for a direct download. Direct downloads are supported for up to 2 gigabytes of data at a time. Alternative approaches for transferring large amounts of data are provided in the Large Data Transfer Methods section of the Data Transfer Guide (https://www.designsafe-ci.org/user-guide/managingdata/datatransfer/#globus).'
+    );
+  });
+
+  // Test that allows downloads of files less than 2 GB
+  it('allows direct file downloads when the file size is below 2 GB', () => {
+    // Mock the dispatch action
+    const mockDispatch = vi.fn();
+    // vi.mock('react-redux', () => ({
+    //   useDispatch: vi.fn(),
+    // }));
+    // useDispatch.mockReturnValue(mockDispatch);
+    // Create a test file whose size is less than 2 GB
+    const testFile = {
+      name: 'test.txt',
+      type: 'file',
+      length: 1000000000,
+      path: '/test.txt',
+    };
+    // Create a spy that watches for the dispatch call
+    vi.spyOn(DataFilesToolbar, 'dispatch').mockImplementationOnce(() => mockDispatch);
```

Comment:
```suggestion
    vi.spyOn(require('react-redux'), 'useDispatch').mockReturnValue(mockDispatch);
```
The above change fixes it and the mock gets called. You need still fix the payload in L337 to make this work.

Reply:
Thanks, Chandra!
I implemented that in my end, and now I'm trying to get the payload to cooperate.

Reply:
Hi, @chandra-tacc!
I've updated the tests and filed out the information above.